src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -1008,13 +1008,15 @@ def _set_up_defaults(
             _get_custom_location_and_extension_id_and_location_from_cluster(cmd, custom_location=custom_location, extension=extension)
             if custom_location.exists:
                 env.custom_location_id = custom_location.get_rid()
+                env.location = custom_location.location
```

Comment:
even --location is provided, you still will use custom location region as environment location?

Reply:
Will use cx input location as env location. Fixed.